lib/client-request.js

```diff
@@ -17,19 +17,14 @@ module.exports = class HTTPClientRequest extends HTTPOutgoingMessage {
     const host = opts.host = opts.host || 'localhost'
     const port = opts.port = opts.port || 80
 
-    const {
-      connection = new HTTPClientConnection(agent.createConnection(opts), opts)
-    } = opts
+    super()
 
-    super(connection.socket)
-
-    connection.req = this
+    agent.addRequest(this, opts)
 
     this.method = method
     this.path = path
     this.headers = { host: host + ':' + port, ...opts.headers }
 
-    this._connection = connection
```

Comment:
This should be initialised to `null` so the shape of the object doesn't change once the connection is assigned in `_onsocket()`. <https://v8.dev/docs/hidden-classes> is great read for more information on that.

Reply:
fixed at https://github.com/holepunchto/bare-http1/pull/15/commits/7b27037279ad23baf14e3075a72356c1715caa42